src/server/GameController.cpp

```diff
@@ -93,32 +115,21 @@ void GameController::updatePlayers(std::vector<client_direction_update> client_d
 /**
     Updates the active status and properties of groups.
 */
-void GameController::refreshAndUpdateGroups() {
+void GameController::updateGroups() {
     // Update groups
     for (auto group : mGroups) {
         group->update();
     }
-
-    handleCollision();
 }
 
-void GameController::handleCollision() {
-    // Get active group and mine circles
-    std::vector<std::shared_ptr<Group>> active_groups = Group::getActiveGroups(mGroups);
-    std::vector<std::shared_ptr<Mine>> active_mines = Mine::getActiveMines(mMines);
-    std::vector<std::shared_ptr<Circle>> active_group_circles = Group::getCircles(active_groups);
-    std::vector<std::shared_ptr<Circle>> active_mine_circles = Mine::getCircles(active_mines);
-    std::vector<std::shared_ptr<Circle>> active_circles;
-
-    // Concat active group and mine circles
-    active_circles.reserve(active_group_circles.size() + active_mine_circles.size());
-    active_circles.insert(
-        active_circles.end(), active_group_circles.begin(), active_group_circles.end());
-    active_circles.insert(
-            active_circles.end(), active_mine_circles.begin(), active_mine_circles.end());
-
-    // Detect and handle collision
-    handle_circle_collision(active_circles);
+void GameController::matchRigid() {
```

Comment:
This step could be done in a cleaner way maybe. Let me know if you have any thoughts.
It can at least have a better name ;)

Reply:
Fixed in b69748c